dashboard/app.R

```diff
@@ -262,55 +278,41 @@ server <- function(input, output, session) {
-    if (input$targetVariable == "Deaths") {
-      signalFilter = DEATH_FILTER
-    }
-    if (input$targetVariable == "Hospitalizations") {
-      signalFilter = HOSPITALIZATIONS_FILTER
-    }
-    filteredScoreDf = scoreDf %>%
-      filter(signal == signalFilter) %>%
-      filter(forecaster %in% input$forecasters)
-
-    if (signalFilter == HOSPITALIZATIONS_FILTER) {
-      filteredScoreDf = filterHospitalizationsAheads(filteredScoreDf)
+  summaryPlot = function(colorSeed = 100, reRenderTruth = FALSE, asOfData = NULL) {
+    filteredScoreDf = filterScoreDf()
+    if (dim(filteredScoreDf)[1] == 0) {
+      output$renderWarningText <- renderText("The selected forecasters do not have enough data to display the selected scoring metric.")
+      return()
     }
-    filteredScoreDf = filteredScoreDf %>% filter(ahead %in% input$aheads)
-    if (input$scoreType == "wis" || input$scoreType == "sharpness") {
-      # Only show WIS or Sharpness for forecasts that have all intervals
-      filteredScoreDf = filteredScoreDf %>% filter(!is.na(`50`)) %>% filter(!is.na(`80`)) %>% filter(!is.na(`95`))
-      if (input$targetVariable == "Deaths") {
-        filteredScoreDf = filteredScoreDf %>% filter(!is.na(`10`)) %>% filter(!is.na(`20`)) %>% filter(!is.na(`30`)) %>%
-                          filter(!is.na(`40`)) %>% filter(!is.na(`60`)) %>% filter(!is.na(`70`)) %>% filter(!is.na(`90`)) %>% filter(!is.na(`98`))
-      }
-      if (dim(filteredScoreDf)[1] == 0) {
-        output$renderWarningText <- renderText("The selected forecasters do not have enough data to display the selected scoring metric.")
-        return()
-      }
-      if (input$scoreType == "wis") {
-        plotTitle = "Weighted Interval Score"
-      }
-      else {
-        plotTitle = "Spread"
+    if (is.null(asOfData)) {
+      if (!is.null(isolate(PREV_AS_OF_DATA())) && dim(isolate(PREV_AS_OF_DATA()))[1] != 0 &&
+          isolate(input$asOf) != '' && isolate(input$asOf) != isolate(MAX_WEEK_END_DATE())) {
+         asOfData = isolate(PREV_AS_OF_DATA())
       }
     }
-    if (input$scoreType == "ae") {
-      plotTitle = "Absolute Error"
-    }
-    if (input$scoreType == "coverage") {
-      plotTitle = "Coverage"
+    if (!is.null(asOfData) && dim(asOfData)[1] != 0) {
+      asOfData = asOfData %>% rename(target_end_date = time_value, as_of_actual = value)
+      asOfData = asOfData[c("target_end_date", "geo_value", "as_of_actual")]
+
+      # Since cases and deaths are shown as weekly incidence, but the "as of" data from the covidcast API
+      # is daily, we need to sum over the days leading up to the target_end_date of each week to get the
+      # weekly incidence.
+      # Get the 'as of' dates that are the target_end_dates in the scoring df
+      dateGroupDf = asOfData %>% filter(asOfData$target_end_date %in% filteredScoreDf$target_end_date)
+      if (dim(dateGroupDf)[1] != 0) {
+        asOfData = filterAsOfData(asOfData, dateGroupDf, filteredScoreDf)
```

Comment:
```suggestion
        # sum over preceding week for all weekly target variables
        asOfData = filterAsOfData(asOfData, dateGroupDf, filteredScoreDf)
```

Reply:
Moved the above detailed comment that explains the summing to here to be more relevant and added this comment to the line in the function that actually performs the summation.